app/actions/handlers.py

```diff
@@ -156,6 +152,16 @@ async def action_pull_events(integration: Integration, action_config: PullEvents
             ]
         )
         for partition in utils.generate_geometry_fragments(geometry_collection=geometry_collection):
+            if partition == "error":
+                msg = f"Error while generating geometry fragments for Geometry collection {geometry_collection}. AOI {aoi_data.id}."
+                await log_activity(
+                    integration_id=integration.id,
+                    action_id="pull_events",
+                    level=LogLevel.WARNING,
+                    title=msg,
+                    data={"message": msg}
+                )
+                break
```

Comment:
Not sure if we need to raise an exception here or just continue @chrisdoehring

Reply:
Thanks @vgarcia13 I'm afraid this as the potential for creating quite a lot of log data we don't want.
I suggest you change this in a few ways:

1. just set a flag in that loop to indicate that a partition is invalid.
2. use that flag _after_ the loop to determine whether to log activity.
3. in the log write a short title (eg. "Errors while generating partitions for Geometry Collection.") and put the data in `data` (ex. ( ```data={"aoi_data": "aoi_data", "geometry_collection": geometry_collection}```)

Go ahead and remove the break -- this is good as a warning and shouldn't abort the execution.

PS. We might find that the geometry collection is sometimes quite large.